ffi/src/lib.rs

```diff
@@ -164,16 +229,19 @@ impl From<Box<[u8]>> for Value {
 #[unsafe(no_mangle)]
 pub unsafe extern "C" fn fwd_free_value(value: *const Value) {
     let value = unsafe { &*value as &Value };
-    if value.len == 0 {
-        return;
+    // We assume that if the length is 0, then the data is a null-terminated string.
+    if value.len > 0 {
+        let recreated_box = unsafe {
+            Box::from_raw(std::slice::from_raw_parts_mut(
+                value.data as *mut u8,
+                value.len,
+            ))
+        };
+        drop(recreated_box);
+    } else {
+        let cstr = unsafe { CString::from_raw(value.data as *mut c_char) };
```

Comment:
Keep the unsafe as small as possible:

```suggestion
        let raw_string_data = value.data as *mut c_char;
        let cstr = unsafe (CString::from_raw(raw_string_data);
```